// @atlaskit/checkbox/dist/es2019/internal/theme.js
import { B100, B200, B300, B400, B50, B75, DN10, DN200, DN30, DN80, DN90, N10, N20, N30, N40, N70, R300 } from '@atlaskit/theme/colors';
const theme = {
  light: {
    borderColor: {
      rest: `var(--ds-border, ${N40})`,
      hovered: `var(--ds-border, ${N40})`,
      disabled: `var(--ds-background-disabled, ${N20})`,
      checked: `var(--ds-background-brand-bold, ${B400})`,
      active: `var(--ds-border, ${B50})`,
      invalid: `var(--ds-border-danger, ${R300})`,
      invalidAndChecked: `var(--ds-border-danger, ${R300})`,
      focused: `var(--ds-border-focused, ${B100})`,
      hoveredAndChecked: `var(--ds-background-brand-bold-hovered, ${B300})`
    },
    boxColor: {
      rest: `var(--ds-background-input, ${N10})`,
      hovered: `var(--ds-surface, ${N30})`,
      disabled: `var(--ds-background-disabled, ${N20})`,
      active: `var(--ds-background-input-pressed, ${B50})`,
      hoveredAndChecked: `var(--ds-background-brand-bold-hovered, ${B300})`,
      checked: `var(--ds-background-brand-bold, ${B400})`
    },
    tickColor: {
      disabledAndChecked: `var(--ds-text-disabled, ${N70})`,
      activeAndChecked: `var(--ds-text-inverse, ${B400})`,
      checked: `var(--ds-text-inverse, ${N10})`
    }
  },
  dark: {
    borderColor: {
      rest: `var(--ds-border, ${DN80})`,
      hovered: `var(--ds-border, ${DN200})`,
      disabled: `var(--ds-background-disabled, ${DN10})`,
      checked: `var(--ds-background-brand-bold, ${B400})`,
      active: `var(--ds-border, ${B200})`,
      invalid: `var(--ds-border-danger, ${R300})`,
      invalidAndChecked: `var(--ds-border-danger, ${R300})`,
      focused: `var(--ds-border-focused, ${B75})`,
      hoveredAndChecked: `var(--ds-background-brand-bold-hovered, ${B75})`
    },
    boxColor: {
      rest: `var(--ds-background-input, ${DN10})`,
      hovered: `var(--ds-surface, ${DN30})`,
      disabled: `var(--ds-background-disabled, ${DN10})`,
      active: `var(--ds-background-input-pressed, ${B200})`,
      hoveredAndChecked: `var(--ds-background-brand-bold-hovered, ${B75})`,
      checked: `var(--ds-background-brand-bold, ${B400})`
    },
    tickColor: {
      disabledAndChecked: `var(--ds-text-disabled, ${DN90})`,
      activeAndChecked: `var(--ds-text-inverse, ${DN10})`,
      checked: `var(--ds-text-inverse, ${DN10})`
    }
  }
};
export default theme;
