// @atlaskit/textarea/dist/esm/component-tokens.js
import { B100, B75, DN10, DN20, DN200, DN30, DN40, DN600, DN90, N0, N10, N100, N20, N30, N40, N70, N900, R400 } from '@atlaskit/theme/colors';
export var disabledBackground = {
  light: "var(--ds-background-disabled, ".concat(N20, ")"),
  dark: "var(--ds-background-disabled, ".concat(DN20, ")")
};
export var disabledBorder = {
  light: "var(--ds-background-disabled, ".concat(N40, ")"),
  dark: "var(--ds-background-disabled, ".concat(DN40, ")")
};
export var invalidBorderColor = {
  light: "var(--ds-border-danger, ".concat(R400, ")"),
  dark: "var(--ds-border-danger, ".concat(R400, ")")
};
export var defaultBorderColor = {
  light: "var(--ds-border, ".concat(N40, ")"),
  dark: "var(--ds-border, ".concat(DN40, ")")
};
export var defaultBorderColorFocus = {
  light: "var(--ds-border-focused, ".concat(B100, ")"),
  dark: "var(--ds-border-focused, ".concat(B75, ")")
};
export var defaultBackgroundColor = {
  light: "var(--ds-background-neutral, ".concat(N10, ")"),
  dark: "var(--ds-background-neutral, ".concat(DN10, ")")
};
export var defaultBackgroundColorHover = {
  light: "var(--ds-surface, ".concat(N30, ")"),
  dark: "var(--ds-surface, ".concat(DN30, ")")
};
export var defaultBackgroundColorFocus = {
  light: "var(--ds-surface, ".concat(N0, ")"),
  dark: "var(--ds-surface, ".concat(DN10, ")")
}; // TODO Subtle hover styles not defined in Figma: https://product-fabric.atlassian.net/browse/DSP-1568

export var subtleHoverBackgroundColor = {
  light: "var(--ds-background-neutral-subtle-hovered, ".concat(N30, ")"),
  dark: "var(--ds-background-neutral-subtle-hovered, ".concat(DN30, ")")
};
export var placeholderTextColor = {
  light: "var(--ds-text-subtlest, ".concat(N100, ")"),
  dark: "var(--ds-text-subtlest, ".concat(DN200, ")")
};
export var textColor = {
  light: "var(--ds-text, ".concat(N900, ")"),
  dark: "var(--ds-text, ".concat(DN600, ")")
};
export var disabledTextColor = {
  light: "var(--ds-text-disabled, ".concat(N70, ")"),
  dark: "var(--ds-text-disabled, ".concat(DN90, ")")
};
export var transparent = {
  light: 'transparent',
  dark: 'transparent'
};
