// @atlaskit/checkbox/dist/cjs/internal/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _colors = require("@atlaskit/theme/colors");

var theme = {
  light: {
    borderColor: {
      rest: "var(--ds-border, ".concat(_colors.N40, ")"),
      hovered: "var(--ds-border, ".concat(_colors.N40, ")"),
      disabled: "var(--ds-background-disabled, ".concat(_colors.N20, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(_colors.B400, ")"),
      active: "var(--ds-border, ".concat(_colors.B50, ")"),
      invalid: "var(--ds-border-danger, ".concat(_colors.R300, ")"),
      invalidAndChecked: "var(--ds-border-danger, ".concat(_colors.R300, ")"),
      focused: "var(--ds-border-focused, ".concat(_colors.B100, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(_colors.B300, ")")
    },
    boxColor: {
      rest: "var(--ds-background-input, ".concat(_colors.N10, ")"),
      hovered: "var(--ds-surface, ".concat(_colors.N30, ")"),
      disabled: "var(--ds-background-disabled, ".concat(_colors.N20, ")"),
      active: "var(--ds-background-input-pressed, ".concat(_colors.B50, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(_colors.B300, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(_colors.B400, ")")
    },
    tickColor: {
      disabledAndChecked: "var(--ds-text-disabled, ".concat(_colors.N70, ")"),
      activeAndChecked: "var(--ds-text-inverse, ".concat(_colors.B400, ")"),
      checked: "var(--ds-text-inverse, ".concat(_colors.N10, ")")
    }
  },
  dark: {
    borderColor: {
      rest: "var(--ds-border, ".concat(_colors.DN80, ")"),
      hovered: "var(--ds-border, ".concat(_colors.DN200, ")"),
      disabled: "var(--ds-background-disabled, ".concat(_colors.DN10, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(_colors.B400, ")"),
      active: "var(--ds-border, ".concat(_colors.B200, ")"),
      invalid: "var(--ds-border-danger, ".concat(_colors.R300, ")"),
      invalidAndChecked: "var(--ds-border-danger, ".concat(_colors.R300, ")"),
      focused: "var(--ds-border-focused, ".concat(_colors.B75, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(_colors.B75, ")")
    },
    boxColor: {
      rest: "var(--ds-background-input, ".concat(_colors.DN10, ")"),
      hovered: "var(--ds-surface, ".concat(_colors.DN30, ")"),
      disabled: "var(--ds-background-disabled, ".concat(_colors.DN10, ")"),
      active: "var(--ds-background-input-pressed, ".concat(_colors.B200, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(_colors.B75, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(_colors.B400, ")")
    },
    tickColor: {
      disabledAndChecked: "var(--ds-text-disabled, ".concat(_colors.DN90, ")"),
      activeAndChecked: "var(--ds-text-inverse, ".concat(_colors.DN10, ")"),
      checked: "var(--ds-text-inverse, ".concat(_colors.DN10, ")")
    }
  }
};
var _default = theme;
exports.default = _default;
