// @atlaskit/checkbox/dist/esm/internal/theme.js
import { B100, B200, B300, B400, B50, B75, DN10, DN200, DN30, DN80, DN90, N10, N20, N30, N40, N70, R300 } from '@atlaskit/theme/colors';
var theme = {
  light: {
    borderColor: {
      rest: "var(--ds-border, ".concat(N40, ")"),
      hovered: "var(--ds-border, ".concat(N40, ")"),
      disabled: "var(--ds-background-disabled, ".concat(N20, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(B400, ")"),
      active: "var(--ds-border, ".concat(B50, ")"),
      invalid: "var(--ds-border-danger, ".concat(R300, ")"),
      invalidAndChecked: "var(--ds-border-danger, ".concat(R300, ")"),
      focused: "var(--ds-border-focused, ".concat(B100, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(B300, ")")
    },
    boxColor: {
      rest: "var(--ds-background-input, ".concat(N10, ")"),
      hovered: "var(--ds-surface, ".concat(N30, ")"),
      disabled: "var(--ds-background-disabled, ".concat(N20, ")"),
      active: "var(--ds-background-input-pressed, ".concat(B50, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(B300, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(B400, ")")
    },
    tickColor: {
      disabledAndChecked: "var(--ds-text-disabled, ".concat(N70, ")"),
      activeAndChecked: "var(--ds-text-inverse, ".concat(B400, ")"),
      checked: "var(--ds-text-inverse, ".concat(N10, ")")
    }
  },
  dark: {
    borderColor: {
      rest: "var(--ds-border, ".concat(DN80, ")"),
      hovered: "var(--ds-border, ".concat(DN200, ")"),
      disabled: "var(--ds-background-disabled, ".concat(DN10, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(B400, ")"),
      active: "var(--ds-border, ".concat(B200, ")"),
      invalid: "var(--ds-border-danger, ".concat(R300, ")"),
      invalidAndChecked: "var(--ds-border-danger, ".concat(R300, ")"),
      focused: "var(--ds-border-focused, ".concat(B75, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(B75, ")")
    },
    boxColor: {
      rest: "var(--ds-background-input, ".concat(DN10, ")"),
      hovered: "var(--ds-surface, ".concat(DN30, ")"),
      disabled: "var(--ds-background-disabled, ".concat(DN10, ")"),
      active: "var(--ds-background-input-pressed, ".concat(B200, ")"),
      hoveredAndChecked: "var(--ds-background-brand-bold-hovered, ".concat(B75, ")"),
      checked: "var(--ds-background-brand-bold, ".concat(B400, ")")
    },
    tickColor: {
      disabledAndChecked: "var(--ds-text-disabled, ".concat(DN90, ")"),
      activeAndChecked: "var(--ds-text-inverse, ".concat(DN10, ")"),
      checked: "var(--ds-text-inverse, ".concat(DN10, ")")
    }
  }
};
export default theme;
